### PR TITLE
feat(server): Server should periodically delete old builds

### PR DESCRIPTION
**Pull request by koh110**
I add cron task that delete old builds on sql to the server.

fixes: https://github.com/GoogleChrome/lighthouse-ci/issues/463

### CODE REVIEW

File: packages/server/src/server.js [hunk 1]
```diff
@@ -55,6 +56,7 @@ async function createApp(options) {
   app.use(errorMiddleware);
 
   startPsiCollectCron(storageMethod, options);
+  startDeletingOldBuildsCron(storageMethod, options);
```
Comment (patrickhulce) [inline packages/server/src/server.js:59]: for consistency with option proposed name `startDeleteOldBuildsCron`?

File: packages/server/src/cron/delete-old-builds.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+'use strict';
```
Comment (patrickhulce) [inline packages/server/src/cron/delete-old-builds.js:1]: could you add a license comment to the top of this file? you can copy/paste from the psi-collect one :)

**Review by patrickhulce — CHANGES_REQUESTED** [review]
thank you @koh110 this is fantastic!! 🎉 

those tests and refactor to share cron utils are great 💯

File: packages/server/src/cron/delete-old-builds.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * @param {number} range
+ * @param {Date} now
+ * @return {Promise<void>}
+ */
+async function deleteOldBuilds(storageMethod, range, now = new Date()) {
+  if (!range || !Number.isInteger(range) || range <= 0) {
+    throw new Error('Invalid range');
+  }
+
+  const runAt = new Date(now.setDate(now.getDate() + range));
```
[the remark this answers is not in the document]
Comment (koh110) [reply packages/server/src/cron/delete-old-builds.js:17]: I agree with your comment. 
> I worry it relies on a bit too much knowledge of range needing to be negative in config

Your code looks to be easier than my code to understand. Thanks for the good suggestions.

I will change it.

File: types/collect.d.ts [hunk 4]
```diff
@@ -44,7 +44,6 @@ declare global {
 
       export interface Options {
         url?: string | string[];
-        autodiscoverUrlBlocklist?: string | string[];
```
[the remark this answers is not in the document]
Comment (koh110) [reply types/collect.d.ts:47]: This is my mistake! I overlooked that. Thanks.

File: types/server.d.ts [hunk 5]
```diff
@@ -173,6 +173,10 @@ declare global {
         sqlConnectionUrl?: string;
         sqlDangerouslyResetDatabase?: boolean;
         sequelizeOptions?: import('sequelize').Options;
+        deleteOldBuilds?: {
```
[the remark this answers is not in the document]
Comment (koh110) [reply types/server.d.ts:176]: This function only works when the storage option is 'sql', so I put it in `StorageOption`.
But, I agree with your comment. I move up to `Options`.

**Review by patrickhulce — APPROVED** [review]
LGTM thank you @koh110! 🎉

**Comment by koh110** [comment]
@patrickhulce Thank you for your review & merge!
I can't wait to use this feature in my lighthouse-ci! Is there anything I can do to help with the release?

**Comment by patrickhulce** [comment]
Nothing in particular, I'll try to ship it out this week 👍

**Comment by koh110** [comment]
I appreciate you! :+1:

**Comment by emmekappa** [comment]
@patrickhulce any idea when this will be released? Is it planned for the `0.6.2`?

**Comment by patrickhulce** [comment]
@emmekappa this has been available since v0.5.1

**Comment by emmekappa** [comment]
@patrickhulce is somehow possible to configure the numbers of days? I cannot see anything in the doc

**Comment by patrickhulce** [comment]
@emmekappa yes it is, it accepts [two options](https://github.com/GoogleChrome/lighthouse-ci/pull/471/files#diff-d8db0ba89c771e37b3a90bea1b84e676ee6b0d0e31f95b5b420c0a83c8932850R187-R190). `maxAgeInDays` is what you're looking for

It wasn't added to the docs yet, but PR welcome :)

**Pull request merged**